sklearn/externals/README

```diff
@@ -1,6 +1,9 @@
 This directory contains bundled external dependencies that are updated
 every once in a while.
 
+Note to developers and advanced users: setting the SKLEARN_SITE_JOBLIB to
+a non null value will force scikit-learn to use the site joblib.
```

Comment:
Maybe note that joblib.Memory cache and joblib pickles may be invalidated when switching from one to the other.